modules/cudf/src/data_frame.ts

```diff
@@ -70,6 +70,89 @@ function _invokeIfNumericSeries<P extends keyof T, T extends TypeMap, R extends
   return Series.new(series._col as Column<R>);
 }
 
+function _concat<DFs extends DataFrame[], T extends TypeMap>(...dfs: DFs): DataFrame {
```

Comment:
We should be able to infer `T` from the combined names in the `DFs` type. If not, we might need to rethink this generic type signature.

```suggestion
function _concat<DFs extends DataFrame[]>(...dfs: DFs): DataFrame {
```

Reply:
Hmm... spent some time with this, but the complexity with typescript types still eludes me. Able to give a little more info?

I'm assuming this is what we want
1. Have some way to keep the types of each of the columns (for every single dataframe passed in)
2. Combine these types together at the end using `&` (similar to join).
3. Typescript should show how many columns result in the concat, and their types.

Assuming what I've said is even possible.